lib/windows/launcher/actions/desktopShortcutActions.js

```diff
@@ -64,6 +65,12 @@ function createAppShortcut(name) {
     };
 }
 
```

Comment:
I see some problems with this file:
- The variables `fileName`, `args`, and `filePath` are defined at the top level and shared among all functions in the file. Try to avoid shared variables like this, as it [makes the application's state unpredictable](https://softwareengineering.stackexchange.com/questions/148108/why-is-global-state-so-evil). Instead, I would suggest creating functions `getShortcutFileName(app)` and `getShortcutArgs(app)` and calling these from `createShortcutFor[Windows|Linux|Mac]()`. The `filePath` is always `config.getDesktopDir()`, so you can remove that variable as well.
- When dragging an app out from the launcher, it looks like it always creates the shortcut on the desktop. A user may drag the shortcut to a different location, such as the start menu. Have you thought about this?
- A `CREATE_APP_SHORTCUT` action is dispatched, but no reducer is handling it. Then there is no need for this action.
- There is a `mode` constant at the top level that is set to `(fs.constants.S_IRWXU | fs.constants.S_IRWXG | fs.constants.S_IRWXO)`. To me it looks like this is permission `777` (read, write, execute for all), is this correct? Seems a bit excessive.

Reply:
Thanks ;)
I will take a loot at it.

Reply:
I have thought about the dragging function, it is easy to detect the position of a cursor inside the window by the js event but it seems not easy to detect which folder the cursor is located outside the window or whether a cursor is located on the startmenu or not. I will keep finding solution for it.
Others have been updated.